### ISSUE TITLE
Unable to send events after update libcurl to v8.1.0

### ISSUE THREAD

**Issue by rodolfoBee** [getsentry/sentry-php#1537]
After updating libcurl to 8.1.0 the Laravel SDK is not able to send events to Sentry, it fails with the following error:

```
[sentry-laravel] error: Failed to send the event to Sentry. Reason: "HTTP/2 stream 1 was reset for "https://o388665.ingest.sentry.io/api/5461612/envelope/".". | {"exception":{},"event":{}}
```

The PHP SDK also fails to send events, but silently. 

https://github.com/curl/curl/releases/tag/curl-8_1_0
https://github.com/curl/curl/releases/tag/curl-8_0_1

Current workaround: downgrade to v8.0.1.

**Comment by krowinski** [comment]
The problem is in ` \Http\Client\Common\Plugin\DecoderPlugin::handleRequest ` as is adding header 'TE' 
`$request = $request->withHeader('TE', $encodings);`

with values
`"TE" => array:3 [
              0 => "gzip"
              1 => "deflate"
              2 => "chunked"
            ]`




And RFC for HTTP/2 specifies `The only exception to this is the TE header field, which MAY be present in an HTTP/2 request; when it is, it MUST NOT contain any value other than "trailers"."`

https://httpwg.org/specs/rfc9113.html#rfc.section.8.2.2

**Comment by mariandomanik** [comment]
Can confirm - either remove TE headers, or switch protocol to 1.1 in

`vendor/symfony/http-client/CurlHttpClient.php`
`public function request`

**Comment by cleptric** [comment]
I opened an issue over at https://github.com/php-http/client-common/

**Comment by mariandomanik** [comment]
When code reaches DecoderPlugin, the protocol is set to 1.1, it is only switched to HTTP2 at CurlHttpClient, after the TE headers are already set

`} elseif (\defined('CURL_VERSION_HTTP2') && (\CURL_VERSION_HTTP2 & CurlClientState::$curlVersion['features']) && ('https:' === $scheme || 2.0 === (float) $options['http_version'])) {
            $curlopts[\CURLOPT_HTTP_VERSION] = \CURL_HTTP_VERSION_2_0;
        }`

In my opinion, the CurlHttpClient should verify it can switch to HTTP2 with the headers, or manage them accordingly after switching to HTTP2, but it is up to debate.

**Comment by cleptric** [comment]
Appreciate the feedback, but let’s move this over to the repository in question.

**Comment by cleptric** [comment]
We released [v3.19.1](https://github.com/getsentry/sentry-php/releases/tag/3.19.1) that enforces HTTP/1.1 if compression is enabled.

**Comment by pawelbaranski** [comment]
if someones stumbles on the issue and upgrade to [v3.19.1](https://github.com/getsentry/sentry-php/releases/tag/3.19.1) is not helping then try to update symfony/http-client to v 5.4.16 at minimum.

`HttplugClient` in earlier version forced using http_version with null value, which results in setting `\CURL_HTTP_VERSION_2_0` as default.

problem
https://github.com/symfony/http-client/blob/v5.4.15/HttplugClient.php#L262
fix
https://github.com/symfony/http-client/blob/v5.4.16/HttplugClient.php#L266
